conf/conda_local_low_memory.config

```diff
@@ -0,0 +1,36 @@
+params {
+    conda_envs_location = "${projectDir}/conda_envs"
+}
+
+conda {
+    enabled = true
+}
+
+process {
+
+    errorStrategy = { task.attempt < 3 ? 'retry' : 'ignore' }
+
+    time = '1h'
+    cpus = 6
+    memory = 4.GB
+
+    withName:
+    'GATK.*|LOFREQ.*|DELLY.*|TBPROFILER.*|MULTIQC.*|FASTQC.*|UTILS.*|FASTQ.*|SAMPLESHEET.*' {
+        //environment does exist:
+        conda = "${params.conda_envs_location}/magma-env-1"
+
+        //environment does NOT exist: the env file contains all the info to create the environment "magma-env-1"
+        //conda = "${params.conda_envs_location}/magma-env-1.yml"
+    }
+
+    withName:
+    'BWA.*|IQTREE.*|SNPDISTS.*|SNPSITES.*|BCFTOOLS.*|BGZIP.*|SAMTOOLS.*|SNPEFF.*|CLUSTERPICKER.*' {
+        //environment does exist:
+        conda = "${params.conda_envs_location}/magma-env-2"
+
+        //environment does NOT exist: the env file contains all the info to create the environment "magma-env-2"
+        //conda = "${params.conda_envs_location}/magma-env-2.yml"
+
+    }
+
+}
```

Comment:
I understand the need for this kind of low-resource profile, however instead of repeating the code from `conda_local` would not be the best way forward.

**Option-1**
Perhaps we can just add another profile for `low_memory` and then upon invoking the pipeline it'd be like 

```console
$ nextflow ... -profile conda_local,low_memory ...
```


Also, the benefit is that we just need to redefine a specific label for within that profile for example 


```
process {

    withLabel: "cpu_8_memory_6" {
        cpus = 6
        memory = 6.GB
    }
}

```

**Option-2**

Define a local config and then provide that at runtime - which avoids changing the code in this repo altogether.


```console
$ ```console
$ nextflow ... -profile conda_local -c low_memory.config  ...
```

Reply:
Hi @abhi18av 

I tried this but got stuck on this:

Unknown method invocation `publishDir` on _parse_closure5 type

Reply:
Hmm, seems like `conda_local_low_memory.config` you created was included 🤔 


I think best to follow the instructions mentioned in https://github.com/TORCH-Consortium/MAGMA?tab=readme-ov-file#customizing-the-pipeline-configuration-for-your-infrastructure and add the `withLabel` snippet to that config  to achieve the final config as shown below

I have added **NOTE** in the comments to clarify how the config override works.

```nextflow


process {
    errorStrategy = { task.attempt < 3 ? 'retry' : 'ignore' }

   //NOTE: Set the baseline defaults for all processes
    time = '1h'
    cpus = 8
    memory = 8.GB


   //NOTE: Use the withName and withLabel to override the baseline defaults 

   withName:FASTQ_VALIDATOR {
      cpus = 2
      memory = 4.GB
   }

   withLabel: "cpu_8_memory_8" {
        cpus = 6
        memory = 6.GB
    }
}

```